Lazy loading in AOT doesn't work in 7.1.0-beta.1

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION YOUR ISSUE MIGHT BE CLOSED WITHOUT INVESTIGATING
-->
### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [X] build
- [X] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions
```
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 7.1.0-beta.1
Node: 11.1.0
OS: darwin x64
Angular: 7.1.0-beta.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.11.0-beta.1
@angular-devkit/build-angular     0.11.0-beta.1
@angular-devkit/build-optimizer   0.11.0-beta.1
@angular-devkit/build-webpack     0.11.0-beta.1
@angular-devkit/core              7.1.0-beta.1
@angular-devkit/schematics        7.1.0-beta.1
@angular/cli                      7.1.0-beta.1
@ngtools/webpack                  7.1.0-beta.1
@schematics/angular               7.1.0-beta.1
@schematics/update                0.11.0-beta.1
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.23.1
    

```

### Repro steps
1) Create a new project
2) Create a lazy module
3) Run an AOT serve or build.

### The log given by the failure
Build succeed but fails during runtime when navigating to the lazy route you will get an error

```
 Error: Uncaught (in promise): Error: Cannot find module './lazy/lazy.module.ngfactory'
Error: Cannot find module './lazy/lazy.module.ngfactory'
```

### Desired functionality
Lazy loading in AOT works.

--- a/packages/angular_devkit/build_angular/test/browser/lazy-module_spec_large.ts
+++ b/packages/angular_devkit/build_angular/test/browser/lazy-module_spec_large.ts
@@ -142,7 +142,7 @@ describe('Browser Builder lazy modules', () => {
       tap((buildEvent) => expect(buildEvent.success).toBe(true)),
       tap(() => {
         expect(host.scopedSync()
-          .exists(join(outputPath, 'lazy-lazy-module.js'))).toBe(true);
+          .exists(join(outputPath, 'lazy-lazy-module-ngfactory.js'))).toBe(true);
       }),
     ).toPromise().then(done, done.fail);
   });
@@ -301,7 +301,7 @@ describe('Browser Builder lazy modules', () => {
     runTargetSpec(host, browserTargetSpec, overrides, DefaultTimeout * 2).pipe(
       tap((buildEvent) => expect(buildEvent.success).toBe(true)),
       tap(() => expect(host.scopedSync()
-        .exists(join(outputPath, 'src-app-lazy-lazy-module.js')))
+        .exists(join(outputPath, 'src-app-lazy-lazy-module-ngfactory.js')))
         .toBe(true)),
     ).toPromise().then(done, done.fail);
   });

--- a/packages/ngtools/webpack/src/angular_compiler_plugin.ts
+++ b/packages/ngtools/webpack/src/angular_compiler_plugin.ts
@@ -696,9 +696,8 @@ export class AngularCompilerPlugin {
               const dependencies = Object.keys(this._lazyRoutes)
                 .map((key) => {
                   const modulePath = this._lazyRoutes[key];
-                  const importPath = key.split('#')[0];
                   if (modulePath !== null) {
-                    const name = importPath.replace(/(\.ngfactory)?\.(js|ts)$/, '');
+                    const name = key.split('#')[0];
 
                     return new this._contextElementDependencyConstructor(modulePath, name);
                   } else {

--- a/packages/ngtools/webpack/src/angular_compiler_plugin.ts
+++ b/packages/ngtools/webpack/src/angular_compiler_plugin.ts
@@ -698,7 +698,7 @@ export class AngularCompilerPlugin {
                   const modulePath = this._lazyRoutes[key];
                   const importPath = key.split('#')[0];
                   if (modulePath !== null) {
-                    const name = importPath.replace(/(\.ngfactory)?(\.(js|ts))?$/, '');
+                    const name = importPath.replace(/(\.ngfactory)?\.(js|ts)$/, '');
 
                     return new this._contextElementDependencyConstructor(modulePath, name);
                   } else {

